controller/replicator.go

```diff
@@ -312,6 +312,95 @@ func (r *replicator) Snapshot(name string, userCreated bool, created string) err
 	return nil
 }
 
+func (r *replicator) GetLastSnapshot() (string, error) {
+	retErrorLock := sync.Mutex{}
+	retError := &BackendError{
+		Errors: map[string]error{},
```

Comment:
don't we need to perform make of this 'map'?

Reply:
make sure we hit error case in testing

Reply:
This is another way to initialize a map. [initialize map in go](https://stackoverflow.com/questions/31064688/which-is-the-nicer-way-to-initialize-a-map-in-golang)